File: dataset/output-bk/OPENAI_gpt-4o/Task16_OPENAI_gpt-4o.java
public class Main {
    public static void main(String[] args) {
        System.out.println(solution("abc", "bc")); // true
        System.out.println(solution("abc", "d")); // false
    }

    public static boolean solution(String str, String ending) {
        return str.endsWith(ending);
    }
}

File: dataset/output-bk/OPENAI_gpt-4o/Task16_OPENAI_gpt-4o.c
#include <stdio.h>
#include <string.h>
#include <stdbool.h>

bool solution(const char *str, const char *ending) {
    size_t str_len = strlen(str);
    size_t ending_len = strlen(ending);
    
    if (ending_len > str_len) {
        return false;
    }
    
    return strcmp(str + str_len - ending_len, ending) == 0;
}

int main() {
    printf("%d
", solution("abc", "bc")); // 1
    printf("%d
", solution("abc", "d")); // 0
    return 0;
}